File: ruby/sparql2.rb
# Ruby querying of the BioPortal triplestore using standard the RDF.rb family of libraries
# Documentation for working with the RDF.rb SPARQL client can be found here: https://github.com/ruby-rdf/sparql-client

require 'rubygems'
require 'sparql/client'

apikey = ""

endpoint = SPARQL::Client.new("http://sparql.bioontology.org/sparql/?apikey=#{apikey}", {method: :get})

query = <<-EOS
PREFIX omv: <http://omv.ontoware.org/2005/05/ontology#>

SELECT ?ont ?name ?acr
WHERE { ?ont a omv:Ontology;
             omv:acronym ?acr;
             omv:name ?name .
} LIMIT 100
EOS

results = endpoint.query(query)

results.each do |statement|
	puts statement.inspect
end

File: python/sparql2.py
""" Simple Python script to query "http://sparql.bioontology.org/sparql/"
    This script uses the SPARQLWrapper Python lib. Download and install from:
        http://sparql-wrapper.sourceforge.net/
"""

from SPARQLWrapper import SPARQLWrapper, JSON, XML, N3, RDF
import pdb

if __name__ == "__main__":
    sparql_service = "http://sparql.bioontology.org/sparql/"

    #To get your API key register at http://bioportal.bioontology.org/accounts/new
    api_key = "YOUR API KEY"

    #Some sample query.
    query_string = """ 
PREFIX omv: <http://omv.ontoware.org/2005/05/ontology#>

SELECT ?ont ?name ?acr
WHERE { ?ont a omv:Ontology;
             omv:acronym ?acr;
             omv:name ?name .
} 
"""
    sparql = SPARQLWrapper(sparql_service)
    sparql.addCustomParameter("apikey",api_key)
    sparql.setQuery(query_string)
    sparql.setReturnFormat(JSON)
    results = sparql.query().convert()
    for result in results["results"]["bindings"]:
        print result["ont"]["value"], result["name"]["value"], result["acr"]["value"]
